Sara/mkt 771 product release notes oct 2023 (#1249)
semgrep/semgrep-docs: docs/release-notes/latest.md

  - Release notes
---

# October 2023 release notes

## 🔧 Semgrep OSS Engine

- The following versions of Semgrep OSS Engine were released in October 2023:
  - [<i class="fas fa-external-link fa-xs"></i> 1.43.0](https://github.com/returntocorp/semgrep/releases/tag/v1.43.0)
  - [<i class="fas fa-external-link fa-xs"></i> 1.44.0](https://github.com/returntocorp/semgrep/releases/tag/v1.44.0)
  - [<i class="fas fa-external-link fa-xs"></i> 1.45.0](https://github.com/returntocorp/semgrep/releases/tag/v1.45.0)
  - [<i class="fas fa-external-link fa-xs"></i> 1.46.0](https://github.com/returntocorp/semgrep/releases/tag/v1.46.0)

## 🌐 Semgrep Cloud Platform

### Added
- Added a button to **Remove** source code manager (SCM) apps. This is helpful when you may have a malconfigured SCM app, such as GitHub's `semgrep-app`, and want to reinstall it. <!--(10688)--> To remove an SCM, click **<i class="fa-solid fa-gear"></i> Settings > Source code managers**.
![Remove your source code manager](/img/settings-scm-remove.png)
- Added Semgrep Assistant to the new onboarding flow. <!--(10716) -->
- **OpenAPI:** Renamed instances of r2c to Semgrep. <!--(10685) -->

> - **Projects page:** Added a new column to display a Semgrep Secrets counter. This counter counts all secrets regardless of validation state. <!--(10588)-->

Reviewer: I wonder if it makes sense to put this under Secrets?

Author: ACKnowledged - will move it.

- **CLI login:** New users are now directed to create a Semgrep org when they are logging in for the first time to Semgrep Cloud Platform from the CLI. <!-- (10596) -->

### Fixed

- Fixed UI issues in the new onboarding flow.
- Fixed an issue where Semgrep Cloud Platform could crash during the onboarding flow. <!--(#10940) -->

> - Various frontend fixes and improvements to the following:

Reviewer: Ah never mind. I see this below

	- Finding detail page
	- Projects page
- Fixed an issue where the **Delete user** functionality did not work for some Semgrep orgs. <!-- (#10756) -->

> - Corrected SARIF output by adding back a missing dict item.

Reviewer: I'm not sure what this means?

Author: There was a missing mapping from Info to Note: [https://github.com/semgrep/semgrep/pull/9077](https://github.com/semgrep/semgrep/pull/9077)

Author: In hindsight - this is actually an OSS fix so I'll remove it.

Reviewer: Ah great!

  - Release notes
---

# October 2023 release notes

## 🔧 Semgrep OSS Engine

- The following versions of Semgrep OSS Engine were released in October 2023:
  - [<i class="fas fa-external-link fa-xs"></i> 1.43.0](https://github.com/returntocorp/semgrep/releases/tag/v1.43.0)
  - [<i class="fas fa-external-link fa-xs"></i> 1.44.0](https://github.com/returntocorp/semgrep/releases/tag/v1.44.0)
  - [<i class="fas fa-external-link fa-xs"></i> 1.45.0](https://github.com/returntocorp/semgrep/releases/tag/v1.45.0)
  - [<i class="fas fa-external-link fa-xs"></i> 1.46.0](https://github.com/returntocorp/semgrep/releases/tag/v1.46.0)

## 🌐 Semgrep Cloud Platform

### Added

> - Added a button to **Remove** source code manager (SCM) apps. This is helpful when you may have a malconfigured SCM app, such as GitHub's `semgrep-app`, and want to reinstall it. <!--(10688)--> To remove an SCM, click **<i class="fa-solid fa-gear"></i> Settings > Source code managers**.

Reviewer 2: ```suggestion
- Added a button to **Remove** source code manager (SCM) apps. This is helpful when you have a misconfigured SCM app, such as GitHub's `semgrep-app`, and want to reinstall it. <!--(10688)--> To remove an SCM, click **<i class="fa-solid fa-gear"></i> Settings > Source code managers**.
```

> ![Remove your source code manager](/img/settings-scm-remove.png)

Reviewer 2: should this image be indented to preserve formatting?

Author: Done. Thank you!

### Changed

- Updated the default CircleCI YAML snippet to include full and diff scans. <!-- (#10678) -->

## 💻 Semgrep Code

### Fixed

- Speed and stability improvements across the product. Semgrep Code pages, such as Findings and Policies, now load faster.
- **Semgrep Assistant:** Component tags are now visible for all Assistant users.
    - **Component tags** use GPT-4 to categorize your code based on its function, such as:
        - Payments
        - User authentication
        - Infrastructure
    - By categorizing your code through component tags, Semgrep Assistant is able to help you prioritize **high-risk issues**, for example if Semgrep has detected a code finding related to payments or user authentication.
    ![Semgrep Assistant Component tag list](/img/assistant-component-tags.png)

## ⛓️ Semgrep Supply Chain

### Added

- Added a new, public [<i class="fas fa-external-link fa-xs"></i> Semgrep Supply Chain API](https://semgrep.dev/api/v1/docs/#tag/SupplyChainService) where you can filter and query third-party vulnerability findings by a variety of parameters, such as:
    - Severity
    - Repository
    - Exposure

> - **C# reachability** is now **GA (generally available)**. Semgrep Supply Chain has added reachability rule support for all C# CVEs from May 2022 onward.

Reviewer 2: ```suggestion
- **C# reachability** is now **GA (generally available)**. Semgrep Supply Chain has had reachability rule support for all C# CVEs since May 2022.
```

Author: Ah this one is tricky - we haven't "had" rule support since May 2022. Instead we just recently added those rules, working backward. Resolve if all good? :)

### Fixed

* **SBOM (software bill of materials) export:** Fixed an issue where SBOM export failed when encountering dependencies with empty names.
* **Vulnerabilities page:** Fixed an issue where triage states did not update until a page refresh. Triage states now update as the user performs a triage action. <!-- (10887) -->

### Changed

- **SBOM export:** The name of the exported SBOM file now follows the following format: `sbom-<org_name>-<repo_name>-<MM-DD-YY_H-m-s>--<serial_number>.<xml|json>` <!-- (10850) -->

## 🔐 Semgrep Secrets (beta)

### Added

- Semgrep Secrets is now in **public beta**.
- **Projects page:** Added a new column to display a Semgrep Secrets counter. This counter counts all secrets regardless of validation state. <!--(10588)-->

### Fixed

- Fixed links to branches in GitLab self-hosted repositories. <!-- (#10897) -->

## 📝 Documentation and knowledge base

### Added
* Added Semgrep Secrets documentation:
	* [<i class="fa-regular fa-file-lines"></i> Conceptual overview of Semgrep Secrets](/semgrep-secrets/conceptual-overview)
	* [<i class="fa-regular fa-file-lines"></i> Getting started with Semgrep Secrets](/semgrep-secrets/getting-started)

> * Added [<i class="fa-regular fa-file-lines"></i> Repository rulesets](/kb/semgrep-ci/github-repository-rulesets-semgrep/) knowledgebase article. This article explains how to scale Semgrep across many GitHub repositories.

Reviewer 2: ```suggestion
* Added [<i class="fa-regular fa-file-lines"></i> Repository rulesets](/kb/semgrep-ci/github-repository-rulesets-semgrep/) knowledge base article. This article explains how to scale Semgrep across many GitHub repositories.
```

Author: I'm pretty sure I should have written a Vale rule for this :) Thank you.

> - Created automated job to sync the help output of the Semgrep CLI tool with CLI reference 

Reviewer 2: ```suggestion
- Created automated job to sync the help output of the Semgrep CLI tool with CLI reference.
```

### Changed

- The Policies documentation has been improved.

### Fixes

> * Various improvements to knowledgebase articles.

Reviewer 2: ```suggestion
* Various improvements to knowledge base articles.
```
